service_container/parent_services.rst

```diff
@@ -117,28 +117,21 @@ duplicated service definitions:
     .. code-block:: php
 
         // config/services.php
+        namespace Symfony\Component\DependencyInjection\Loader\Configurator;
+
         use App\Repository\DoctrineUserRepository;
         use App\Repository\DoctrinePostRepository;
         use App\Repository\BaseDoctrineRepository;
-        use Symfony\Component\DependencyInjection\ChildDefinition;
-        use Symfony\Component\DependencyInjection\Reference;
-
-        $container->register(BaseDoctrineRepository::class)
-            ->setAbstract(true)
-            ->addArgument(new Reference('doctrine.orm.entity_manager'))
-            ->addMethodCall('setLogger', [new Reference('logger')])
-        ;
 
-        // extend the App\Repository\BaseDoctrineRepository service
-        $definition = new ChildDefinition(BaseDoctrineRepository::class);
-        $definition->setClass(DoctrineUserRepository::class);
-        $container->setDefinition(DoctrineUserRepository::class, $definition);
-
-        $definition = new ChildDefinition(BaseDoctrineRepository::class);
-        $definition->setClass(DoctrinePostRepository::class);
-        $container->setDefinition(DoctrinePostRepository::class, $definition);
-
-        // ...
+        return function(ContainerConfigurator $configurator) {
+            $configurator->services()
```

Comment:
```php
$container->services()
    // NOTE: this must be set before a "defaults()" call so that it can be extended
    ->set(BaseDoctrineRepository::class)->abstract()
    ->args([ref('doctrine.orm.entity_manager')])
    ->call('setLogger', [ref('logger')->nullOnInvalid()])

    ->set(DoctrineUserRepository::class)->parent(BaseDoctrineRepository::class)
    ->set(DoctrinePostRepository::class)->parent(BaseDoctrineRepository::class)

    // ...
;
```
?